php-checks/src/main/java/org/sonar/php/checks/formatting/ControlStructureSpacingCheck.java

```diff
@@ -19,88 +19,157 @@
  */
 package org.sonar.php.checks.formatting;
 
-import com.sonar.sslr.api.AstNode;
-import com.sonar.sslr.api.Token;
+import com.google.common.collect.ImmutableList;
 import org.sonar.php.api.PHPKeyword;
 import org.sonar.php.api.PHPPunctuator;
 import org.sonar.php.checks.FormattingStandardCheck;
-import org.sonar.php.parser.PHPGrammar;
+import org.sonar.php.checks.utils.TokenVisitor;
+import org.sonar.php.tree.impl.PHPTree;
+import org.sonar.plugins.php.api.tree.ScriptTree;
+import org.sonar.plugins.php.api.tree.Tree;
+import org.sonar.plugins.php.api.tree.Tree.Kind;
+import org.sonar.plugins.php.api.tree.lexical.SyntaxToken;
+import org.sonar.plugins.php.api.tree.statement.ForEachStatementTree;
+import org.sonar.plugins.php.api.visitors.PHPSubscriptionCheck;
 
-public class ControlStructureSpacingCheck extends SpacingCheck {
+import java.util.Iterator;
+import java.util.List;
+
+public class ControlStructureSpacingCheck extends PHPSubscriptionCheck implements FormattingCheck {
+
+  private static final String CONTROL_STRUCTURES_KEYWORD_MESSAGE = "between this \"%s\" keyword and the opening %s";
+  private static final String FOR_SEMICOLON_MESSAGE = "Put exactly one space after each \";\" character in the \"for\" statement.";
+  private static final String FOREACH_MESSAGE = "Put exactly one space after and before %s in \"foreach\" statement.";
+  private static final Kind[] CONTROL_STRUCTURES = {
+    Kind.IF_STATEMENT,
+    Kind.ELSEIF_CLAUSE,
+    Kind.ELSE_CLAUSE,
+    Kind.DO_WHILE_STATEMENT,
+    Kind.WHILE_STATEMENT,
+    Kind.FOR_STATEMENT,
+    Kind.FOREACH_STATEMENT,
+    Kind.SWITCH_STATEMENT,
+    Kind.TRY_STATEMENT,
+    Kind.CATCH_BLOCK
+  };
+
+  private FormattingStandardCheck check;
+
+  @Override
+  public void checkFormat(FormattingStandardCheck formattingCheck, ScriptTree scriptTree) {
+    this.check = formattingCheck;
+    this.scanTree(scriptTree);
+  }
 
   @Override
-  public void visitNode(FormattingStandardCheck formattingCheck, AstNode node) {
-    if (formattingCheck.isOneSpaceBetweenKeywordAndNextToken && node.is(FormattingStandardCheck.getControlStructureNodes())) {
-      checkSpaceBetweenKeywordAndNextNode(formattingCheck, node);
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.copyOf(CONTROL_STRUCTURES);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (check.isOneSpaceBetweenKeywordAndNextToken && tree.is(CONTROL_STRUCTURES)) {
```

Comment:
@vilchik-elena indeed it looks like it is useless, might be a left over from refactoring.